source/iface/address-space.rst

```diff
@@ -2,211 +2,1105 @@
+
+::
+
+  template <sycl::access::address_space Space,
+            sycl::access::decorated DecorateAddress,
+            typename ElementType>
+  multi_ptr<ElementType, Space, DecorateAddress>
+  address_space_cast(ElementType* pointer);
+
+Global function to create a ``sycl::multi_ptr`` instance
+from ``pointer``, using the address space and decoration
+specified via the ``Space`` and ``DecorateAddress`` template arguments.
+
+An implementation must return ``nullptr`` if the run-time value of
+``pointer`` is not compatible with ``Space``, and must issue a
+compile-time diagnostic if the deduced address space for
+``pointer`` is not compatible with ``Space``.
+
+``make_ptr``
+------------
+
+::
+
+  template <typename ElementType,
+            sycl::access::address_space Space,
+            sycl::access::decorated DecorateAddress>
+  multi_ptr<ElementType, Space, DecorateAddress>
+  make_ptr(ElementType* pointer);
+
+Deprecated in SYCL 2020. Use `address_space_cast`_ instead.
+
+Global function to create a ``sycl::multi_ptr`` instance
+depending on the address space of the ``pointer`` argument.
+An implementation must return ``nullptr`` if the run-time value
+of ``pointer`` is not compatible with ``Space``, and must issue
+a compile-time diagnostic if the deduced address space is not
+compatible with ``Space``.
+
+Operators
+=========
+
+``operator=``
+-------------
+
+.. rubric:: Overload 1
+
+::
+
+  sycl::multi_ptr& operator=(const sycl::multi_ptr&);
+
+Copy assignment operator.
+
+.. rubric:: Overload 2
+
+::
+
+  sycl::multi_ptr& operator=(sycl::multi_ptr&&);
+
+Move assignment operator.
+
+.. rubric:: Overload 3
+
+::
+
+  sycl::multi_ptr& operator=(std::nullptr_t);
+
+Assigns ``nullptr`` to the ``sycl::multi_ptr``.
+
+.. rubric:: Overload 4
 
 ::
 
-     friend ElementType& operator*(const sycl::multi_ptr& mp);
+  template <sycl::access::address_space AS,
+            sycl::access::decorated IsDecorated>
+  sycl::multi_ptr&
+  operator=(const sycl::multi_ptr<value_type, AS, IsDecorated>&);
+
+.. note::
+
+  Available only when:
+  ::
+
+    (Space == sycl::access::address_space::generic_space &&
+     AS != sycl::access::address_space::constant_space)
+
+Assigns the value of the left hand side
+``sycl::multi_ptr`` into the ``sycl::generic_ptr``.
+
+.. rubric:: Overload 5
+
+::
+
+  template<sycl::access::address_space AS,
+           sycl::access::decorated IsDecorated>
+  sycl::multi_ptr&
+  operator=(sycl::multi_ptr<value_type, AS, IsDecorated>&&);
+
+.. note::
+
+  Available only when:
+  ::
+
+    (Space == sycl::access::address_space::generic_space &&
+     AS != sycl::access::address_space::constant_space)
+
+Move the value of the left hand side
+``sycl::multi_ptr`` into the ``sycl::generic_ptr``.
+
+
+``operator[]``
+--------------
+
+::
+
+  reference operator[](std::ptrdiff_t i) const;
+
+Available only when: ``!std::is_void_v<value_type>``.
+
+Returns a reference to the ``i``th pointed
+value. The value ``i`` can be negative.
```

Comment:
The formatting here is broken in the rendered doc

Reply:
Fixed with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/109/commits/81dbb1b7fe4ce81bbb486cf20599c9459a09671f